### DISCUSSION TITLE
Preserve synced tab selection across pages/sessions

### DISCUSSION THREAD

**Discussion by bhollis** [withastro/starlight#1851] [Feature Requests]
### What version of `starlight` are you using?

0.22.1

### What is your idea?

I'm extracting this comment so it doesn't get lost now that the original feature request (synced Tabs) is done: https://github.com/withastro/starlight/discussions/953#discussioncomment-9160332

It would be great if we could allow synced tabs to remember their state between sessions (reloads, page navigation). This would mean that tabs with a `syncKey` set would immediately swap to the tab corresponding to the last chosen sync key at page load / component load. The state could be stored in `localStorage`.

### Why is this feature necessary?

When building code documentation, we have sample code available in many different languages. It's likely that a reader only cares about one of those at a time. Once they select their preferred language in one place, we'd like for that to be applied to every code sample. As it is now, they have to make their selection again every time they load the page.

### Do you have examples of this feature in other projects?

- https://docusaurus.io/docs/markdown-features/tabs#syncing-tab-choices

### Participation

- [ ] I am willing to submit a pull request for this feature.

**Comment by delucis** [comment]
Thanks for creating a feature request to track this @bhollis! We’re currently exploring how `localStorage` relates to to GDPR/PECR requirements for a feature like this. Once we have a clear picture of what our duties are there, I think we’ll be able to give a definitive answer on how we could add this.

**Reply by bhollis** [reply]
Just checking in - do you have a sense of how GDPR/PECR would have anything to say about this, since localStorage isn't communicated with any third party?

**Reply by delucis** [reply]
The main concerns were with some details in PECR specifically, I forget the details now.

We’ve decided we’re comfortable adding this feature based on our research, although obviously folks will need to evaluate for themselves if that meets whatever regulatory standards they need to follow. Always happy to hear feedback on this.

Draft PR for synced tabs is here: https://github.com/withastro/starlight/pull/2087

**Comment by HiDeoo** [comment]
Starting with Starlight [`v0.26.0`](https://github.com/withastro/starlight/releases/tag/%40astrojs%2Fstarlight%400.26.0), [synced `<Tabs>`](https://starlight.astro.build/components/tabs/#sync-tabs) are now persistent so that a user's choices are reflected across page navigations.
